src/validations/rules/sar.sch

```diff
@@ -175,6 +181,215 @@
+                name="first-result"
+                value="//oscal:result[xs:dateTime(oscal:start) eq max(//oscal:result/oscal:start ! xs:dateTime(.))]/@uuid" />
+            <!-- In-scope controls -->
+            <sch:let
+                name="in-scope-controls"
+                value="$ssp-doc//oscal:implemented-requirement/@control-id" />
+            <!-- Baseline controls -->
+            <sch:let
+                name="baseline-controls"
+                value="$profile-doc//oscal:alter/@control-id" />
+            <!-- in scope controls (in SSP) that exist in the baseline controls -->
+            <sch:let
+                name="combined"
+                value="$in-scope-controls[. = $baseline-controls]" />
+            <!-- In scope controls (in SSP) that do not exist in baseline -->
+            <sch:let
+                name="not-in-baseline-controls"
+                value="$in-scope-controls[not(. = $baseline-controls)]" />
+            <!-- profile alter elements that have a @control-id value that matches a value in the $combined list (indicating the control is in scope for the SSP) -->
+            <sch:let
+                name="matching-control-ids"
+                value="
+                    for $i in $combined
+                    return
+                        $profile-doc//oscal:alter[@control-id = $i]/oscal:add[contains(@by-id, '_obj.')][oscal:prop[@name = 'response-point']]/@by-id" />
+            <!-- targetid values of the findings of the most recent result -->
+            <sch:let
+                name="first-result-finding-target-IDs"
+                value="//oscal:result[@uuid = $first-result]//oscal:finding//oscal:target[@type = 'objective-id']/@target-id" />
+            <!-- matching profile control ids with first result finding target ids -->
+            <sch:let
+                name="target-IDs-combined"
+                value="$matching-control-ids[. = $first-result-finding-target-IDs]" />
+            <!-- profile control ids that are not in the first result finding target ids -->
+            <sch:let
+                name="target-IDs-combined-missing"
+                value="$matching-control-ids[not(. = $first-result-finding-target-IDs)]" />
+            <sch:assert
+                diagnostics="no-duplicate-target-id-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.3"
+                id="no-duplicate-target-id"
+                role="error"
+                test="
+                    if (oscal:finding[parent::oscal:result[@uuid = $first-result]])
+                    then
+                        count(oscal:finding/oscal:target[@type = 'objective-id']/@target-id) = count(distinct-values(oscal:finding/oscal:target[@type = 'objective-id']/@target-id))
+                    else
+                        true()">All objective targets in the most recent result assembly must be distinct.</sch:assert>
+        </sch:rule>
```

Comment:
Rather prolix. Presuming the (unstated in the documentation afaict) objective is "no duplicate result targets" (in any result, not just the most recent), this can be replaced with
```
<sch:rule
    context="oscal:target">
    <sch:assert
        role="error"
        test="count(ancestor::oscal:result//oscal:target/@target-id[. eq current()/@target-id]) eq 1">Every finding target is
        unique within a result.</sch:assert>
</sch:rule>
```
and the assertion placed in the subsequent identical context rather than having two identical contexts. As well as obviating a lot of unnecessary code, it allows a failed assertion to precisely identify the offending `target` as opposed to provoking vague unease at the `result` level.

The documentation reference should identify "_Guide to OSCAL-based FedRAMP Security Assessment **Results** (SAR)_".